internal/core/command/device_test.go

```diff
@@ -42,101 +46,132 @@ var (
 	mismatch = "d200-c200-mismatch"
 	d200c404 = "d200-c404"
 	d200c500 = "d200-c500"
+	d200c200 = "d200c200"
 
 	TestCommandId = "TestCommandID"
 )
 
+func createTestDeviceWithPathUrl(commandID string, deviceID string) models.Device {
+
+	return models.Device{
+		AdminState: models.Unlocked,
+		Service: models.DeviceService{
+			Addressable: models.Addressable{
+				Path: "/api/v1/device/" + deviceID + "/command/" + commandID}}}
+}
+
+func createDeviceRequestWithPathParameters(
+	sampleDevice models.Device,
+	params map[string]string) *http.Request {
+
+	req, _ := http.NewRequest(http.MethodGet, cmdURI, nil)
+	req.URL.Path = sampleDevice.Service.Addressable.Path
+
+	return mux.SetURLVars(req, params)
+}
+
 // commandByDeviceID
 func TestExecuteGETCommandByDeviceIdAndCommandId(t *testing.T) {
 	tests := []struct {
 		name        string
-		deviceId    string
-		commandId   string
+		status      string
+		request     *http.Request
 		expectedErr error
 	}{
 		{
 			"Device-InvalidObjectId",
 			status400,
-			TestCommandId,
+			createDeviceRequestWithPathParameters(
+				createTestDeviceWithPathUrl(TestCommandId, status400),
+				map[string]string{ID: status400, COMMANDID: TestCommandId}),
```

Comment:
All of these tests specify `ID` as `statusxxx` to test invalid data. Can we use a better named variable or string that states that it is invalid for clarity? 

This applies to the other test arguments for this test and the associated mocks.